src/lib.rs

```diff
@@ -1390,6 +1392,85 @@ impl Default for StashFlags {
     }
 }
 
+macro_rules! define_attr_inspector {
+    ($enum_name:ident, $enum_doc:literal, $fn_name:ident, $fn_doc:literal, $s:ty) => {
+        #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+        #[doc = $enum_doc]
+        pub enum $enum_name<'string> {
+            /// The attribute is set to true.
+            True,
+            /// The attribute is unset (set to false).
+            False,
+            /// The attribute is set to a string.
+            String(&'string $s),
+            /// The attribute is not specified.
+            Unspecified,
+        }
+
+        #[doc = $fn_doc]
+        pub fn $fn_name(attr_value: Option<&$s>) -> $enum_name<'_> {
+            match unsafe {
+                raw::git_attr_value(attr_value.map_or(ptr::null(), |v| v.as_ptr().cast()))
+            } {
+                raw::GIT_ATTR_VALUE_TRUE => $enum_name::True,
+                raw::GIT_ATTR_VALUE_FALSE => $enum_name::False,
+                raw::GIT_ATTR_VALUE_STRING => $enum_name::String(attr_value.unwrap()),
+                raw::GIT_ATTR_VALUE_UNSPECIFIED => $enum_name::Unspecified,
+                _ => unreachable!(),
+            }
+        }
+    };
+}
+
+define_attr_inspector!(
+    AttrValue,
+    "All possible states of an attribute, using [`prim@str`] to represent the string.\n\n\
+     This enum is returned by [`attr_value`].",
+    attr_value,
+    "Returns the state of an attribute by inspecting its [value](Repository::get_attr) \
+     by a [string](prim@str).",
+    str
+);
+
+/// Converts [`AttrValueBytes`] to [`AttrValue`]. If the attribute is [set to a string](`AttrValueBytes::String`),
+/// this implementation will use [`str::from_utf8`] to perform the conversion.
+impl<'string> TryFrom<AttrValueBytes<'string>> for AttrValue<'string> {
```

Comment:
Instead of having two enums, could there perhaps be one enum with a string/bytes variant where the bytes are guaranteed to always be invalid utf-8?

Reply:
> where the bytes are guaranteed to always be invalid utf-8?

Does that mean `AttrValue::new()` will always perform UTF-8 validation? Could I add another `new` that skips UTF-8 validation?

Reply:
Seems reasonable to me!